rootpy/stats/dataset.py

```diff
@@ -13,9 +13,31 @@
     'DataSet',
 ]
 
-
 class DataSet(NamedObject, QROOT.RooDataSet):
     _ROOT = QROOT.RooDataSet
+    class Entry(object):
+        def __init__(self, idx, dataset):
+            self.idx_ = idx
+            self.dataset_ = dataset
+            
+        @property
+        def leaves(self):
+            return asrootpy(self.dataset_.get(self.idx_))
+        
+        @property
+        def weight(self):
+            self.dataset_.get(self.idx_) #set current event
```

Comment:
``self.dataset_.get(self.idx_)`` could be moved to ``Entry.__init__`` to avoid having it called twice when getting leaves and weight.